### PR TITLE
Add variable to set file to add to the startup script

### LINKED ISSUE #3: Allow for additional custom startup script.
We need the ability to deploy a security agent for monitoring the datalab VMs.  Can there be a way to specify a custom https://github.com/terraform-google-modules/terraform-google-datalab/blob/master/modules/template_files/templates/startup_script.tpl or an additional scrip when the instance is created?

### PR DESCRIPTION
**Pull request by paulpalamarchuk**
* Fix #3

### CODE REVIEW

**Comment by nick4fake** [comment]
@paulpalamarchuk 
image
Looks like it recreates instance every time. I'm still rechecking it, but this PR might be the reason.

File: modules/template_files/outputs.tf [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 output "startup_script" {
   description = "Rendered start script from template"
-  value       = data.template_file.startup_script.rendered
+  value       = "${data.template_file.startup_script.rendered}${data.local_file.additional_startup_script[0].content}"
```
[the remark this answers is not in the document]
Comment (nick4fake) [reply modules/template_files/outputs.tf]: Please update this one:
image

[the remark this answers is not in the document]
Comment (nick4fake) [reply modules/template_files/outputs.tf]: image
Please update it

File: modules/default_instance/variables.tf [hunk 2]
```diff
@@ -102,3 +102,9 @@ variable "datalab_docker_image" {
   description = "Datalab docker image to use"
   default     = "gcr.io/cloud-datalab/datalab:latest"
 }
+
+variable "append_to_startup_script" {
+  default     = ""
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply modules/default_instance/variables.tf]: changed to `null`

File: modules/gpu_instance/variables.tf [hunk 3]
```diff
@@ -112,3 +112,9 @@ variable "gpu_type" {
   description = "The GPU type for the Datalab instance"
   default     = "nvidia-tesla-k80"
 }
+
+variable "append_to_startup_script" {
+  default     = ""
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply modules/gpu_instance/variables.tf]: changed to `null`

File: modules/template_files/outputs.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply modules/template_files/outputs.tf]: updated

**Comment by paulpalamarchuk** [comment]
@nick4fake I've discovered behaviour. Looks like a bug.
Created [issue 4824](https://github.com/terraform-providers/terraform-provider-google/issues/4824)

File: modules/template_files/main.tf [hunk 4]
```diff
@@ -14,6 +14,10 @@
  * limitations under the License.
  */
 
+locals {
+  startup_script_content = var.append_to_startup_script == null ? data.template_file.startup_script.rendered : "${data.template_file.startup_script.rendered}${data.local_file.additional_startup_script[0].content}"
```
Comment (aaron-lane) [inline modules/template_files/main.tf]: We should include newline characters to ensure that the two scripts do not collide.

```suggestion
  startup_script_content = var.append_to_startup_script == null ? data.template_file.startup_script.rendered : "${data.template_file.startup_script.rendered}\n\n${data.local_file.additional_startup_script[0].content}"
```

**Pull request merged**